File: app/src/main/java/com/example/fang/b16traveldomain/model/dataresource/UserDataSource.java
package com.example.fang.b16traveldomain.model.dataresource;

import android.content.Context;

import com.example.fang.b16traveldomain.model.dataresource.localdataresource.LocalUserResource;
import com.example.fang.b16traveldomain.model.dataresource.localdataresource.SharedPreference;

public class UserDataSource implements UserRepository{
    LocalUserResource mLocalUserResource;

    public UserDataSource(Context context) {
        this.mLocalUserResource = new SharedPreference(context);
    }

    @Override
    public void saveUser() {
        mLocalUserResource.saveUser();
    }

    @Override
    public void saveDate(String date) {
        mLocalUserResource.saveDate(date);

    }

    @Override
    public void getUser() {
        mLocalUserResource.getUser();

    }

    @Override
    public void getDate() {
        mLocalUserResource.getDate();

    }

    @Override
    public void saveRoute(String route) {
        mLocalUserResource.saveRoute(route);
    }

    @Override
    public String getRoute() {
        return mLocalUserResource.getRoute();
    }
}

File: app/src/main/java/com/example/fang/b16traveldomain/model/dataresource/UserDataSource.kt
package com.example.fang.b16traveldomain.model.dataresource;

import android.content.Context;

import com.example.fang.b16traveldomain.model.dataresource.localdataresource.LocalUserResource;
import com.example.fang.b16traveldomain.model.dataresource.localdataresource.SharedPreference;

class UserDataSource(context : Context) : UserRepository{
    lateinit var  mLocalUserResource:LocalUserResource

    init {
        this.mLocalUserResource = SharedPreference (context)
    }

    override fun saveUser() {
        mLocalUserResource.saveUser()
    }

    override fun saveDate(date : String) {
        mLocalUserResource.saveDate(date)
    }

    override fun getUser() {
        mLocalUserResource.getUser()
    }

    override fun getDate() : String{
        return mLocalUserResource.getDate()
    }

    override fun saveRoute(route:String) {
        mLocalUserResource.saveRoute(route)
    }

    override fun getRoute() :String{
        return mLocalUserResource.getRoute()
    }
}
